.github/workflows/ci.yml

```diff
@@ -0,0 +1,119 @@
+name: ci
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+"
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@755da8c3cf115ac066823e79a1e1788f8940201b # v3.2.0
+        with:
+          fetch-depth: 0
+
+      - name: Setup Go
+        uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
+        with:
+          go-version: "1.19"
+          cache: true
+
+      - name: ci/check-style
+        run: make check-style
+
+      - name: ci/check-modules
+        run: make check-modules
+
+  test:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@755da8c3cf115ac066823e79a1e1788f8940201b # v3.2.0
+        with:
+          fetch-depth: 0
+          path: src/github.com/mattermost/mattermost-operator
+
+      - name: Setup Go
+        uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
+        with:
+          go-version: "1.19"
+          cache: true
+          cache-dependency-path: src/github.com/mattermost/mattermost-operator/go.sum
+
+      - name: Generate Operator Manifests
+        run: |
+          cd ${GITHUB_WORKSPACE}/src/github.com/mattermost/mattermost-operator
+
+          make clean
+          make operator-sdk
+
+          mkdir -p /tmp/apis/mattermost/v1alpha1
```

Comment:
mkdirs could be concantanated in a single line with :

mkdir -p /tmp/apis/mattermost/v1alpha1 /tmp/apis/mattermost/v1beta1 /tmp/config/crd/bases

Less syscalls !